frontend/src/pages/modelRegistrySettings/ModelRegistryTableRowStatus.tsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+
+import { Label, Popover, Stack, StackItem } from '@patternfly/react-core';
+import {
+  CheckCircleIcon,
+  DegradedIcon,
+  ExclamationCircleIcon,
+  ExclamationTriangleIcon,
+  InProgressIcon,
+} from '@patternfly/react-icons';
+
+import { K8sCondition } from '~/k8sTypes';
+
+enum ModelRegistryStatus {
+  Progressing = 'Progressing',
+  Degraded = 'Degraded',
+  Available = 'Available',
+  IstioAvailable = 'IstioAvailable',
+  GatewayAvailable = 'GatewayAvailable',
+}
+
+enum ModelRegistryStatusLabel {
+  Progressing = 'Progressing',
+  Available = 'Available',
+  Degrading = 'Degrading',
+  Unavailable = 'Unavailable',
+}
+
+enum ConditionStatus {
+  True = 'True',
+  False = 'False',
+}
+
+interface ModelRegistryTableRowStatusProps {
+  conditions: K8sCondition[] | undefined;
+}
+
+export const ModelRegistryTableRowStatus: React.FC<ModelRegistryTableRowStatusProps> = ({
+  conditions,
+}) => {
+  const conditionsMap =
+    conditions?.reduce((acc: Record<string, K8sCondition | undefined>, condition) => {
+      acc[condition.type] = condition;
+      return acc;
+    }, {}) ?? {};
+  let statusLabel: string = ModelRegistryStatusLabel.Progressing;
+  let icon = <InProgressIcon />;
+  let color: React.ComponentProps<typeof Label>['color'] = 'blue';
+  let popoverMessages: string[] = [];
+  let popoverTitle = '';
+
+  if (Object.values(conditionsMap).length) {
+    const {
+      [ModelRegistryStatus.Available]: availableCondition,
+      [ModelRegistryStatus.Progressing]: progressCondition,
+      [ModelRegistryStatus.Degraded]: degradedCondition,
+    } = conditionsMap;
+    const lastAvailableConditionTime = new Date(
+      availableCondition?.lastTransitionTime ?? '',
+    ).getTime();
+
+    popoverMessages =
+      availableCondition?.status === ConditionStatus.False
+        ? Object.values(conditionsMap).reduce((messages: string[], condition) => {
+            if (condition?.status === ConditionStatus.False && condition.message) {
+              messages.push(condition.message);
+            }
+
+            return messages;
+          }, [])
+        : [];
+
+    // Available
+    if (availableCondition?.status === ConditionStatus.True) {
+      statusLabel = ModelRegistryStatusLabel.Available;
+      icon = <CheckCircleIcon />;
+      color = 'green';
+    }
+    // Progressing
+    else if (
+      progressCondition?.status === ConditionStatus.True &&
+      lastAvailableConditionTime <= new Date(progressCondition.lastTransitionTime ?? '').getTime()
```

Comment:
Testing this on the modelregistry-ui cluster, the registry "Dallas MR" has a Progressing condition that is the exact same time as its Available:False condition, and it's been stuck that way for days. I think we should change this `<=` to a `<`, we only want something to take precedence over Available:False if it's actually newer. Same goes for the similar comparison on line 91 for Degrading.